engine/lib/generic_printer/generic_printer.ml

```diff
@@ -430,10 +430,17 @@ module Make (F : Features.T) (View : Concrete_ident.VIEW_API) = struct
           separate_map comma print#generic_param >> group >> angles
 
         method arm' : arm' fn =
-          fun { arm_pat; body } ->
+          fun { arm_pat; body; guard } ->
             let pat = print#pat_at Arm_pat arm_pat |> group in
             let body = print#expr_at Arm_body body in
-            pat ^^ string " => " ^^ body ^^ comma
+            let g =
+              match guard with
+              | Some { guard = IfLet { lhs; rhs; _ }; _ } ->
+                  string " if let " ^^ print#pat_at Arm_pat lhs ^^ string " = "
+                  ^^ print#expr_at Arm_body rhs
+              | None -> string ""
```

Comment:
Can you:
 - use `empty` instead of `string ""`
 - use a `Option.map ~f:(...) guard |> Option.value ~default:empty`?

Reply:
Oh and why not renaming `g` into `guard` here?